### DISCUSSION TITLE
Why doesn't React Router allow accessing navigate, location, params through the router object?

### DISCUSSION THREAD

**Discussion by arn4v** [remix-run/react-router#9401] [General]
Why doesn't React Router allow accessing `NavigateFunction`, params, and location objects globally?

In many instances, router state needs to be accessed/mutated outside of React but since router state/navigate function is only accessible through hooks it leads to this weird prop/argument drilling of navigate/location/params to non-React functions.

Why can't location, params and navigate be accessed through the router object returned by `createBrowserRouter`?

**Comment by supersnager** [comment]
@arn4v Try router.state & router.navigate
